skops/io/_utils.py

```diff
@@ -1,4 +1,135 @@
 import importlib
+from functools import (  # type: ignore
+    GenericAlias,
+    _find_impl,
```

Comment:
I'm somehow more comfortable using private sklearn methods rather than private python methods lol. Let's see how much headache this would give us.

Reply:
I didn't check the implementations in detail, but in the worst case, we could probably copy the code of `_find_impl` here. I wouldn't do it now since it depends on another private function which in turn depends on another private function :D